src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,97 @@
-export const Pagination = () => {};
+import { getNumbers, getPaginatedItems } from '../../utils';
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  setCurrentPage: (arg: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage = 1,
+  setCurrentPage,
+}: Props) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
```

Comment:
The `pages` array is being used to determine the number of pages based on `total` and `perPage`, but it's not necessary to generate an array of numbers for this purpose. You can directly calculate the number of pages and use that value instead, to optimize memory usage.